src/gui/application.cpp

```diff
@@ -183,6 +184,13 @@ Application::Application(int &argc, char **argv)
     , _debugMode(false)
     , _backgroundMode(false)
 {
+    // Work around a bug in KDE's qqc2-desktop-style which breaks
+    // buttons with icons not based on a name, by forcing a style name
+    // the platformtheme plugin won't try to force qqc2-desktops-style
+    // anymore.
+    // Can be removed once the bug in qqc2-desktop-style is gone.
+    QQuickStyle::setStyle("Default");
```

Comment:
I honestly fail to understand the logic behind moving that code here instead of just changing the order in `main.cpp`
so why ?

Reply:
from https://bugreports.qt.io/browse/QTBUG-57522 it looks like we just have to modify the order of lines to get the application to be built before

Reply:
setStyle() must be done before one loads any qml stuff. If I move that to just after OCC::Application the icons in the main dialog are not displayed properly.

Reply:
> from https://bugreports.qt.io/browse/QTBUG-57522 it looks like we just have to modify the order of lines to get the application to be built before

That is the case for QApplication -- but here you are using a OCC::Application, and its constructor have calls to qml.

https://github.com/nextcloud/desktop/blob/master/src/gui/application.cpp#L330
https://github.com/nextcloud/desktop/blob/master/src/gui/owncloudgui.cpp#L68